geode-core/src/main/java/org/apache/geode/management/internal/cli/shell/unsafe/GfshSignalHandler.java

```diff
@@ -80,9 +81,11 @@ protected void handleDefault(final sun.misc.Signal sig, final ConsoleReader cons
     }
   }
 
-  protected SignalHandler getOriginalSignalHandler(final Signal signal) {
+  private SignalHandler getOriginalSignalHandler(final Signal signal) {
     final SignalHandler handler = originalSignalHandlers.get(signal);
-    return (handler == SignalHandler.SIG_DFL || handler == SignalHandler.SIG_IGN ? null : handler);
+    // return (handler == SignalHandler.SIG_DFL || handler == SignalHandler.SIG_IGN ? null :
```

Comment:
Commented out code. And this changed the behavior, is the new behavior correct? I'm not sure what this was doing.

Reply:
The original code relied on the fact that the internal API handle method would return the default handlers for some events if they weren't ever initialized. Setting then back to `null` to the default has the same effect in the internal API, so this logic was really unnecessary. With the new wrapper API the original default is preserved and reset rather than nulled. Still producing the same effect.

Reply:
Scratch that... It looks like we were trying to chain singles here but not for these two defaults. You're right this will result in a change in behavior, for which we have no test for. Let me work out an integration test if possible for this. Not sure we can given the nature of the signals.

Reply:
And only if it is the default signal handler on these two signals.